src/store.rs

```diff
@@ -610,18 +670,98 @@ pub struct LevelCacheStore<E: Element> {
     // unnecessarily.
     store_size: usize,
 
+    // If provided, the store will use this method to access base
+    // layer data.
+    reader: Option<ExternalReader<R>>,
+
     _e: PhantomData<E>,
 }
 
-impl<E: Element> ops::Deref for LevelCacheStore<E> {
+impl<E: Element, R: Read + Send + Sync> fmt::Debug for LevelCacheStore<E, R> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("LevelCacheStore")
+            .field("len", &self.len)
+            .field("elem_len", &self.len)
+            .field("data_width", &self.data_width)
+            .field("cache_index_start", &self.cache_index_start)
+            .field("store_size", &self.store_size)
+            .finish()
+    }
+}
+
+impl<E: Element, R: Read + Send + Sync> LevelCacheStore<E, R> {
+    // Used for opening v2 compacted DiskStores.
+    pub fn new_from_disk_with_reader(
+        store_range: usize,
+        config: &StoreConfig,
+        reader: Option<ExternalReader<R>>,
```

Comment:
this should not be an option, as the point of this method is to take an `ExternalReader`